docs/guides/tooling/typescript-support.mdx

```diff
@@ -71,6 +71,28 @@ If that does not work, try restarting the IDE.
 
 :::
 
+### Processing your Cypress configuration and plugins
+
+Cypress needs to be able to transpile your Cypress configuration and plugins written in TypeScript in order to make them executable within Cypress' Node.js runtime.
+To do this, Cypress will attempt to read the user's TypeScript and project configuration to apply the correct TypeScript loader to Cypress' Node.js runtime.
+
+If your project is an `ESM` package (short for [ECMAScript Module](https://nodejs.org/api/esm.html#modules-ecmascript-modules)), Cypress attempts
+to apply the [ts-node/esm](https://github.com/TypeStrong/ts-node?tab=readme-ov-file#esm) Node.js loader to resolve the Cypress configuration and plugins. `ESM` is determined by Cypress if you have the `type: "module"` key-value pair present inside your project's `package.json`.
+
+Otherwise, regular [ts-node](https://github.com/TypeStrong/ts-node?tab=readme-ov-file#node-flags-and-other-tools) is required into Cypress' Node.js runtime.
+Since Node.js by itself can only interpret CommonJS files, Cypress attempts to make your TypeScript configuration compatable with Cypress' Node.js runtime.
+To do this, Cypress overrides the following configuration values found inside your project's `tsconfig.json`:
+
+```json
+{
+  "module": "commonjs",
+  "moduleResolution": "node",
+  "preserveValueImports": false // only applicable to TypeScript 4.5 and greater
```

Comment:
@AtofStryker Someone left a note on a json example in our docs recently about how if you copy/paste it, it will be invalid due to the comments. I don't know if we want to update this so people don't copy/past the comment...........

Reply:
thats actually a good point. ill update it

Reply:
I just removed it since most are on typescript 4.5 and above and likely can infer if the setting is applied or not based on their version